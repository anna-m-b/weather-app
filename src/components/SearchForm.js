import React from 'react';
import PropTypes from 'prop-types';
import '../styles/SearchForm.css';
import '../styles/normalize.css';

const SearchForm = ({ onInputChange, onSubmit, searchInput }) => {
  return (
    <form className="SearchForm" onSubmit={onSubmit} data-testid="search-form">
      <p className="SearchForm_instruction">
        Search for a UK city to see its weather forecast
      </p>
      <span className="SearchForm_search">
        <label htmlFor="search">
          <input
            type="text"
            value={searchInput}
            placeholder="Enter a city"
            id="search"
            data-testid="search-box"
            onChange={onInputChange}
          />
        </label>
        <button type="submit" data-testid="search-button">
          Search
        </button>
      </span>
    </form>
  );
};

SearchForm.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  searchInput: PropTypes.string.isRequired,
};

export default SearchForm;
