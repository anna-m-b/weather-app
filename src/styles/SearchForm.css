
.SearchForm {
  /* display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center; */
  grid-column: 1 / 3;
  display: grid;
  justify-content: center;
  width: 100%;
  padding-top: 10px;
  padding-bottom: 10px;
  /* border-bottom: 1px solid rgba(142, 142, 142, 0.728);
  border-top: 1px solid rgba(142, 142, 142, 0.728); */
  margin: 10px 0 10px 0;

}

.SearchForm_search {
  width: 100%;
  display: flex;
  justify-content: center;
  align-content: center;
}

.SearchForm input {
  height: 25px;
  font-size: 0.7rem;
  width: 300px;
  margin: 5px;
}

.SearchForm button {
  height: 30px;
  font-size: 0.8rem;
  margin-top: 0.2rem;
  width: 100px;
  margin-top: 6px;
  border: none;
  background-color: rgba(0, 0, 0, 0.762);
  color: white;
  border-radius: 5px;
}

.SearchForm_instruction {
  font-size: 14px;
  font-style: italic;
  margin: 0 0 5px 0;
}

@media screen and (max-width: 600px) {
  .SearchForm_instruction {
    font-size: 0.7rem;
  }
  .SearchForm button {
    height: 10px;
    width: 60px;
    
  }

  .SearchForm input {
    height: 10px;
    font-size: 0.7rem;
    width: 150px;
    margin: 5px;
  }

}