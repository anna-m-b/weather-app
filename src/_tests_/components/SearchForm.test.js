import React from 'react';
import { render, screen } from '@testing-library/react';
import SearchForm from '../../components/SearchForm';

describe('SearchForm', () => {
  const validProps = {
    onInputChange: () => {},
    onSubmit: () => {},
    searchInput: 'Manchester',
  };

  it('renders correctly', () => {
    const { asFragment } = render(<SearchForm {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('button', () => {
    it('renders the button with text "search"', () => {
      render(<SearchForm {...validProps} />);
      expect(screen.getByTestId('search-button')).toHaveTextContent('Search');
    });
  });
});
