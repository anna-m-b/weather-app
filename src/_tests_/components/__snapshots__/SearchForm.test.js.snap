// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="SearchForm"
  >
    <label
      for="search"
    >
      <input
        id="search"
        placeholder="Enter a city"
        type="text"
        value="Manchester"
      />
    </label>
    <button
      data-testid="search-button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
